workers/unity/Assets/Gdk/Core/Error/ConnectionError.cs

```diff
@@ -0,0 +1,20 @@
+namespace Improbable.Gdk.Core
+{
+    public enum ConnectionErrorReason
+    {
+        CannotEstablishConnection,
+        DeploymentNotFound,
+        InvalidConfig
+    }
+
+    public struct ConnectionError : IErrorContext
```

Comment:
using structs results in boxing for every single `HandleError` call. Would it be better to use classes here? @samiwh as he is more familiar with boxing rules then I am

Reply:
It will box if it needs to convert to an object (which seems like it will given the use of an interface)

Reply:
It will however this is an infrequently called api (also boxing almost always has to happen on actually logging objects). 
We could try allocating a class from a pool in future if we find this is creating an appreciable amount of garbage. I would also expect an IL2CPP compiler to optimise this out.

Reply:
TL;DR I don't think this is a performance bottleneck, but it's easy to revisit if we need to. Could also make this just a struct and have other meta data in a user defined context obejct, which will be null by default. That way nothing would be unboxed (and possibly boxed), but it would be a worse API.